src/main/java/com/atlassian/jira/cloud/jenkins/config/ConfigManagementLink.java

```diff
@@ -0,0 +1,319 @@
+
+import javax.inject.Inject;
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+import static com.atlassian.jira.cloud.jenkins.util.IpAddressProvider.getIpAddress;
+import static com.atlassian.jira.cloud.jenkins.Config.ATLASSIAN_API_URL;
+
+@Extension
+public class ConfigManagementLink extends ManagementLink
+        implements Describable<ConfigManagementLink> {
+
+    private static final Logger LOGGER = Logger.getLogger(ConfigManagementLink.class.getName());
+    private transient PluginConfigApi pluginConfigApi;
+    private transient SecretRetriever secretRetriever;
+    private JiraCloudPluginConfig config;
+    private Category category;
+
+    @Inject
+    public void setPluginConfigApi(final PluginConfigApi pluginConfigApi) {
+        this.pluginConfigApi = pluginConfigApi;
+    }
+
+    @Inject
+    public void setSecretRetriever(final SecretRetriever secretRetriever) {
+        this.secretRetriever = secretRetriever;
+    }
+
+    @DataBoundConstructor
+    public ConfigManagementLink() {
+        super();
+        this.config = JiraCloudPluginConfig.get();
+        this.category = Category.CONFIGURATION;
+    }
+
+    @Override
+    public Descriptor<ConfigManagementLink> getDescriptor() {
+        return new DescriptorImpl();
+    }
+
+    public Category getCategory() {
+        return category;
+    }
+
+    @Override
+    public String getIconFileName() {
+        return null;
+        // TODO - use this assert instead when we want to expose link
+        // return "notepad.png";
+    }
+
+    @Override
+    public String getUrlName() {
+        return "atlassian-jira-software-cloud";
+    }
+
+    @Override
+    public String getDisplayName() {
+        return null;
+        // TODO - use this assert instead when we want to expose link
+        // return "Atlassian Jira Software Cloud";
+    }
+
+    public void doIndex(final StaplerRequest req, final StaplerResponse res) {
+        try {
+            req.setAttribute("config", this.config);
+            res.setContentType("text/html");
+            req.getView(this, "/com/atlassian/jira/cloud/jenkins/configuration/config.jelly")
+                    .forward(req, res);
+        } catch (Exception e) {
+            LOGGER.severe(e.getMessage());
+        }
+    }
+
+    public String getRegexFromFormData(final JSONObject formData, final String key) {
+        if (!formData.has(key)) {
+            return "";
+        }
+        return formData.getJSONObject(key).optString(key + "Regex", "");
+    }
+
+    @VisibleForTesting
+    void SendConfigDataToJira(final JSONObject formData) {
+        JSONArray sitesArray = formData.getJSONArray("sites");
+
+        String ipAddress = getIpAddress();
+
+        boolean autoBuildsEnabled = formData.has("autoBuilds");
```

Comment:
Instead of string literals, can we convert all these to enums? e.g. `private static final String AUTO_BUILDS = "autoBuilds";`